packages/aws-cdk-lib/aws-rds/lib/instance.ts

```diff
@@ -171,14 +173,25 @@ export abstract class DatabaseInstanceBase extends Resource implements IDatabase
     // Get ISecurityGroup from securityGroupId
     let securityGroups: ec2.ISecurityGroup[] = [];
     const dbsg: string[] = instance.DBSecurityGroups;
-    if (dbsg) {
+    if (dbsg && dbsg.length > 0) {
       securityGroups = dbsg.map(securityGroupId => {
         return ec2.SecurityGroup.fromSecurityGroupId(
           scope,
           `LSG-${securityGroupId}`,
           securityGroupId,
         );
       });
+    } else {
+      const vpcsg: string[] = instance.VPCSecurityGroups;
+      if (vpcsg && vpcsg.length > 0) {
+        securityGroups = vpcsg.map(securityGroupId => {
+          return ec2.SecurityGroup.fromSecurityGroupId(
+            scope,
+            `LSG-${securityGroupId}`,
+            securityGroupId,
+          );
+        });
+      }
```

Comment:
This needs the mutual exclusivity validation logic (mentioned [here](https://github.com/aws/aws-cdk/pull/34906/files#r2251234232)) as well. While the current logic ensures only one group is used, a user who specifies both groups would not understand this without reading the implementation.

```suggestion
    // Validation
    if (instance.DBSecurityGroups && instance.VPCSecurityGroups) { 
       // throw an error explaining that only one of them can be specified
    }
    
    const sg: string[] = instance.VPCSecurityGroups || instance.DBSecurityGroups || [];
    securityGroups = dbsg.map(securityGroupId => {
        return ec2.SecurityGroup.fromSecurityGroupId(
          scope,
          `LSG-${securityGroupId}`,
          securityGroupId,
        );
      });
```

Reply:
Another question: would this benefit from validation of the security group ID format i.e. DB groups start with `dbsg-`, while VPC groups start with `sg-`?

Not sure how the CDK code will fail if the security group ID is invalid.

Reply:
> EC2-Classic has already been retired

I missed this nuance - I think you are correct that it should not matter.

Reply:
I'd say this is still valid, purely from a code maintainability point of view:

```
const sg: string[] = instance.VPCSecurityGroups || instance.DBSecurityGroups || [];
securityGroups = sg.map(securityGroupId => {
   return ec2.SecurityGroup.fromSecurityGroupId(
      scope,
      `LSG-${securityGroupId}`,
      securityGroupId,
    );
});
```

Since we're confident that only either one will be provided, and they're both string arrays, reusing the logic is safe.

Reply:
Looks good, thanks!